src/vs/base/browser/ui/highlightedlabel/highlightedLabel.ts

```diff
@@ -49,15 +49,48 @@ export class HighlightedLabel implements IDisposable {
 		this.render();
 	}
 
+	private adjustHighlightsForEscapedLineBreaks(highlights: IHighlight[]): IHighlight[] {
```

Comment:
Hm, the separating is a little off. One part escapes new lines and the other part accounts for it... What if my label already contains escaped newlines and what if the highlights have been computed correctly accounting for that?

In stead, you should do all of this in the `set`-method, e.g. `set(text: string, highlights: IHighlight[] = [], escapeNewLines: boolean = false)`. So that you know when to adjust highlights and when not to adjust them.